### PR TITLE
Webhook for updating recurring charges

### PR DESCRIPTION
**Pull request by cadecairos**
When a charge succeeds, it will hit this endpoint, which will update the charge with it's subscription's metadata and a relevant description

### CODE REVIEW

File: routes/stripe.js [hunk 1]
```diff
@@ -73,5 +73,16 @@ module.exports = {
...
+  updateCharge: function(chargeId, updateData, callback) {
+    stripe.charges.update(chargeId, updateData, callback);
+  },
+  retrieveCharge: function(chargeId, callback) {
+    stripe.charges.retrieve(chargeId, {
+      expand: ["invoice"]
+    }, callback);
+  },
+  retrieveSubscription: function(customerId, subscriptionId, callback) {
+    stripe.customers.retrieveSubscription(customerId, subscriptionId, callback);
```
Comment (ScottDowne) [inline routes/stripe.js]: I'm fine with or without these wrappers around stripe.etc

File: server.js [hunk 2]
```diff
@@ -282,6 +282,13 @@ module.exports = function(options) {
         auth: "stripe"
       },
       handler: routes['stripe-dispute']
+    }, {
+      method: "POST",
+      path: "/stripe/charge-succeeded",
```
Comment (ScottDowne) [inline server.js]: I wonder if it's wise to put webhook routes behind something like `webhook/stripe/charge-succeeded/dispute-callback/etc`?

File: routes/stripe.js (hunk 1 above)
Comment (cadecairos) [inline routes/stripe.js]: I'd prefer not to do it, but I don't want to re-instantiate stripe in the index.js file. I suppose I could just export the stripe object here.

File: routes/index.js [hunk 3]
```diff
@@ -415,6 +415,55 @@ var routes = {
...
+  },
+  'stripe-charge-succeeded': function(request, reply) {
+    var event = request.payload;
+    var charge = event.data.object;
+
+    if (event.type !== 'charge.succeeded') {
+      return reply('This hook only processes charge succeeded events');
+    }
+
+    stripe.retrieveCharge(
```
Comment (cadecairos) [inline routes/index.js]: I make this call to expand the invoice object, which then gives me the subscription id, which I can request.

File: server.js (hunk 2 above)
Comment (cadecairos) [inline server.js]: The other webhook doesn't follow this convention, so I think we'd want to handle that change in a new issue if we think it's warranted

File: routes/stripe.js (hunk 1 above)
Comment (ScottDowne) [inline routes/stripe.js]: Ah, I see.

Hm. I say let's just leave it as you have it for right now. Not worth worrying about it right now imo.

**Pull request merged**